jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/UpgradeResponse.java

```diff
@@ -88,57 +79,4 @@
      * @return the status code
      */
     int getStatusCode();
-
-    /**
-     * Issue a forbidden upgrade response.
-     * <p>
-     * This means that the websocket endpoint was valid, but the conditions to use a WebSocket resulted in a forbidden
-     * access.
-     * <p>
-     * Use this when the origin or authentication is invalid.
-     *
-     * @param message the short 1 line detail message about the forbidden response
-     * @throws IOException if unable to send the forbidden
-     */
-    void sendForbidden(String message) throws IOException;
-
-    /**
-     * Set the accepted WebSocket Protocol.
-     *
-     * @param protocol the protocol to list as accepted
-     */
-    void setAcceptedSubProtocol(String protocol);
-
-    /**
-     * Set the list of extensions that are approved for use with this websocket.
-     * <p>
-     * Notes:
-     * <ul>
-     * <li>Per the spec you cannot add extensions that have not been seen in the {@link UpgradeRequest}, just remove
-     * entries you don't want to use</li>
-     * <li>If this is unused, or a null is passed, then the list negotiation will follow default behavior and use the
-     * complete list of extensions that are
-     * available in this WebSocket server implementation.</li>
-     * </ul>
-     *
-     * @param extensions the list of extensions to use.
-     */
-    void setExtensions(List<ExtensionConfig> extensions);
-
-    /**
-     * Set a header
-     * <p>
-     * Overrides previous value of header (if set)
-     *
-     * @param name the header name
-     * @param value the header value
-     */
-    void setHeader(String name, String value);
-
-    /**
-     * Set the HTTP Response status code
-     *
-     * @param statusCode the status code
-     */
-    void setStatusCode(int statusCode);
```

Comment:
Yeah ... this api interface being this useless is also a point of concern for me.

Reply:
Why is it now useless? it still has all the getters.

I could combine this interface with the Server extension of it, but all these methods on the client side would just be implemented with `UnsupportedOperationException`.

The same issue goes for the setters on the `ClientUpgradeRequest` which are not in the `UpgradeRequest` interface, as they would be unimplementable on the server side.